### PR TITLE
PMM-2891 add docker image building for pmm-client

### PR DESCRIPTION
**Pull request by delgod**

### CODE REVIEW

File: build/pmm-client-docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+FROM centos:latest
```
Comment (delgod) [inline build/pmm-client-docker/Dockerfile]: it is openshift business requirement to use rhel based image.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-10e4648
client docker - perconalab/pmm-client-fb:HEAD-10e4648
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-10e4648.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-6b62768
client docker - perconalab/pmm-client-fb:PR-87-6b62768
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-6b62768.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-6c28b43
client docker - perconalab/pmm-client-fb:PR-87-6c28b43
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-6c28b43.tar.gz

File: build/pmm-client-docker/Dockerfile (hunk 1 above)
Comment (EvgeniyPatlan) [reply build/pmm-client-docker/Dockerfile]: Maybe it is better to use centos:7 instead of latest?

Comment (delgod) [reply build/pmm-client-docker/Dockerfile]: has sense, fixed.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-660e29f
client docker - perconalab/pmm-client-fb:PR-87-660e29f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-660e29f.tar.gz

File: build/pmm-client-docker/Dockerfile-alpine [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+COPY pmm-client.tar.gz /tmp/pmm-client.tar.gz
+RUN tar -zxpf /tmp/pmm-client.tar.gz -C /tmp \
+    && cd /tmp/pmm-client-* \
+    && bash ./install
+
+
+FROM alpine:latest
+RUN apk add --update --no-cache bash
+RUN mkdir /lib64 && ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2
```
Comment (arvenil) [inline build/pmm-client-docker/Dockerfile-alpine]: Can you comment in the source code why this is needed?

File: build/pmm-client-docker/Dockerfile-alpine [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ENV MONIT_VERSION=5.25.2
+
+# Compile and install monit
+RUN apk add --update make gcc curl zlib-dev libressl-dev musl-dev \
+    && cd /tmp \
+    && curl -sS https://mmonit.com/monit/dist/monit-${MONIT_VERSION}.tar.gz | gunzip -c - | tar -xf - \
+    && cd monit-${MONIT_VERSION} \
+    && ./configure --without-pam \
+    && make \
+    && make install
```
Comment (arvenil) [inline build/pmm-client-docker/Dockerfile-alpine]: Isn't monit already in alpine repo? Why to compile it? https://pkgs.alpinelinux.org/packages?name=monit&branch=v3.8

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-074ce80
client docker - perconalab/pmm-client-fb:PR-87-074ce80
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-074ce80.tar.gz

File: build/pmm-client-docker/entrypoint.sh [hunk 4]
```diff
@@ -3,6 +3,37 @@
...
+wait_for_port() {
+    local HOST=$1
+    local PORT=$2
+
+    for i in `seq 6` ; do
+        nc -z "${HOST}" "${PORT}" > /dev/null 2>&1 && result=$? || result=$?
+        if [ $result -eq 0 ] ; then
+            return
+        fi
+        sleep 10
```
Comment (arvenil) [inline build/pmm-client-docker/entrypoint.sh]: I don't like such large waiting steps. How long it takes currently, on average, to pass those checks? Is that less than 10s but just sometimes more? Or it's more close to 50s on average?

Comment (delgod) [reply build/pmm-client-docker/entrypoint.sh]: usualy 10s is enoght

File: build/pmm-client-docker/Dockerfile-alpine (hunk 3 above)
Comment (delgod) [reply build/pmm-client-docker/Dockerfile-alpine]: reworked, thanks

File: build/pmm-client-docker/Dockerfile-alpine (hunk 2 above)
Comment (delgod) [reply build/pmm-client-docker/Dockerfile-alpine]: added

File: build/pmm-client-docker/Dockerfile-alpine [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    && cd /tmp/pmm-client-* \
+    && bash ./install
+
+
+FROM alpine:latest
+RUN apk add --update --no-cache bash curl
+
+# golang binaries typically have only one dynamic library dependency - /lib64/ld-linux-x86-64.so.2
+# but Alpine Linux uses a smaller realization
+RUN mkdir /lib64 && ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2
```
Comment (AlekSi) [inline build/pmm-client-docker/Dockerfile-alpine]: No-no-no. `ld-linux` is not a libc and not even a real library. pmm-client should not depend on any libc, and using musl can cause problems.
Let's figure out why is this needed at all. Wild guess: we are not compiling it with `CGO_ENABLED=0` environment variable.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-b639d55
client docker - perconalab/pmm-client-fb:PR-87-b639d55
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-b639d55.tar.gz

File: build/pmm-client-docker/entrypoint.sh (hunk 4 above)
Comment (delgod) [reply build/pmm-client-docker/entrypoint.sh]: reworked to make 1s checks

File: build/pmm-client-docker/Dockerfile-alpine (hunk 5 above)
Comment (delgod) [reply build/pmm-client-docker/Dockerfile-alpine]: problem gone with CGO_ENABLED=0

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-cd503e6
client docker - perconalab/pmm-client-fb:PR-87-cd503e6
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-cd503e6.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-87-7dddd7a
client docker - perconalab/pmm-client-fb:PR-87-7dddd7a
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm-client/pmm-client-PR-87-7dddd7a.tar.gz
